tests/linked_list_test.c

```diff
@@ -0,0 +1,223 @@
+/*
+ *  Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+#include <aws/common/linked_list.h>
+#include <aws_test_harness.h>
+
+struct int_value {
+    struct aws_linked_list_node node;
+    int value;
+};
+
+static int test_linked_list_order_push_back_pop_front() {
+    struct aws_linked_list list;
+
+    static size_t list_size = 4;
+    aws_linked_list_init(&list);
+
+    struct int_value first = (struct int_value){.value = 1};
+    struct int_value second = (struct int_value){.value = 2};
+    struct int_value third = (struct int_value){.value = 3};
+    struct int_value fourth = (struct int_value){.value = 4};
+
+    ASSERT_INT_EQUALS(0, aws_linked_list_length(&list), "List size should be 0.");
+
+    aws_linked_list_push_back(&list, (struct aws_linked_list_node *)&first);
```

Comment:
`aws_linked_list` is feeling a bit verbose to me. Can/should we abbreviate our list names? `aws_alist` (array), `aws_dlist` (doubly linked intrusive) for example.